### PR TITLE
Migrate format test to ES Module

### PR DESCRIPTION
**Pull request by fisker**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
This is ridiculous, how the tests passed on Windows, but failed on MacOS and Linux?

**Comment by fisker** [comment]
Finally, this is working, thanks to the runner from babel project, it's also much faster.

**Comment by fisker** [comment]
@duailibe @sosukesuzuki  Can you review this?

**Review by duailibe — APPROVED** [review]

File: tests/format/angular/angular/jsfmt.spec.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-run_spec(__dirname, ["angular"], { trailingComma: "none" });
-run_spec(__dirname, ["angular"]);
-run_spec(__dirname, ["angular"], { printWidth: 1 });
-run_spec(__dirname, ["angular"], { htmlWhitespaceSensitivity: "ignore" });
-run_spec(__dirname, ["angular"], { bracketSpacing: false });
+run_spec(import.meta, ["angular"], { trailingComma: "none" });
```
Comment (sosukesuzuki) [inline tests/format/angular/angular/jsfmt.spec.js:1]: Why `run_spec` receive `import.meta`? Isn't `import.meta.url` is enough?

File: tests/config/jest-light-runner/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@babel/jest-light-runner",
```
Comment (sosukesuzuki) [inline tests/config/jest-light-runner/package.json:2]: This is not under `@babel`
```suggestion
  "name": "jest-light-runner",
```

File: jest.config.mjs [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import path from "node:path";
 import createEsmUtils from "esm-utils";
-import installPrettier from "./tests/config/install-prettier.mjs";
+import installPrettier from "./tests/config/install-prettier.js";
 
+// @ts-ignore
```
Comment (sosukesuzuki) [inline jest.config.mjs:5]: I prefer `@ts-expect-error`. (or fix it)
```suggestion
// @ts-expect-error
```

**Review by sosukesuzuki — COMMENTED** [review]
Thank you for great work!! Please check comments.

File: tests/format/angular/angular/jsfmt.spec.js (hunk 1 above)
Comment (fisker) [reply tests/format/angular/angular/jsfmt.spec.js:1]: `import.meta` might get useful properties we might want to use in the future, and it's shorter.

File: tests/config/jest-light-runner/package.json (hunk 2 above)
Comment (fisker) [reply tests/config/jest-light-runner/package.json:2]: Maybe you missed this readme file https://github.com/fisker/prettier/blob/esm-format-test/tests/config/jest-light-runner/readme.md

It's copied from babel, let's not do unnecessary change, so we can keep sync with the one in babel repo.

Comment (sosukesuzuki) [reply tests/config/jest-light-runner/package.json:2]: I saw the README but It seemed a bit high-context that a package named `@babel/...` is defined by Prettier.

If there are no changes between all to the jest-light-runner implemented in Babel and Prettier, can we have a script to synchronize it? And if the README says something like the following about the script, I think it's OK.

> Do not update this package manually. This package will only be updated by the script.

File: tests/format/angular/angular/jsfmt.spec.js (hunk 1 above)
Comment (sosukesuzuki) [reply tests/format/angular/angular/jsfmt.spec.js:1]: I got it.

File: tests/config/jest-light-runner/package.json (hunk 2 above)
Comment (sosukesuzuki) [reply tests/config/jest-light-runner/package.json:2]: Of course, that work can be done in other PRs.

Comment (fisker) [reply tests/config/jest-light-runner/package.json:2]: I've made some changes... I don't have strong feeling for the package name, I can change it if you insist. 

One thing you should know, these files even not following the code style, I turned eslint and prettier off on them.

Comment (sosukesuzuki) [reply tests/config/jest-light-runner/package.json:2]: If there are any changes, the name of the package should not start with `@babel/`.

Btw, how did you change the package?

Comment (fisker) [reply tests/config/jest-light-runner/package.json:2]: > Btw, how did you change the package?

https://github.com/prettier/prettier/pull/11902/commits/baabfda5fcad07057f1b7fd417b9c2f076d1500a#diff-0688392cb01e5b47f4f1249386e343f205a7390c18506421c2442616a22a2fddR29

L29-L36 was added.

Comment (sosukesuzuki) [reply tests/config/jest-light-runner/package.json:2]: Thanks.

Comment (fisker) [reply tests/config/jest-light-runner/package.json]: https://github.com/prettier/prettier/pull/11902/commits/172153127b6134f7bd9ce52f77774c9409eae191

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**